core/services/workflows/syncer/workflow_registry.go

```diff
@@ -221,22 +254,32 @@ func (w *workflowRegistry) Start(_ context.Context) error {
 				return
 			}
 
-			w.lggr.Debugw("Loading initial workflows for DON", "DON", don.ID)
-			loadWorkflowsHead, err := w.loadWorkflows(ctx, don, reader)
-			if err != nil {
-				// TODO - this is a temporary fix to handle the case where the chainreader errors because the contract
-				// contains no workflows.  To track: https://smartcontract-it.atlassian.net/browse/CAPPL-393
-				if !strings.Contains(err.Error(), "attempting to unmarshal an empty string while arguments are expected") {
-					w.lggr.Errorw("failed to load workflows", "err", err)
-					return
-				}
+			// Start goroutines to gather changes from Workflow Registry contract
+			switch w.syncStrategy {
+			case SyncStrategyEvent:
+				w.wg.Add(1)
+				go func() {
+					defer w.wg.Done()
+					w.syncUsingEventStrategy(ctx, don, reader)
+				}()
+			case SyncStrategyReconciliation:
+				w.syncUsingReconciliationStrategy(ctx, don, reader)
```

Comment:
_nit_ prefer to use a convention of creating the go routines in one place.  any reason why event strategy is wrapped but not reconciliation strategy?  If it's about reading from the event loop, that can be started as a go routine before the switch statement.

Reply:
It happened because the reconciliation strategy starts 2 loops wrapped in go routines and the event strategy starts one loop that needs to be wrapped. Originally both of these were wrapped on the top level here, but Cedric pointed out that we should run as little go routines as possible.

I just moved the wrapping here deeper into `w.syncUsingEventStrategy`. See if you like that stylistically. Still doesn't bring all goroutines to ones place, but does make it read consistently.

If you feel strongly about having all goroutines in one place I'll have to get rid of sync strategy handlers in favor of handlers for each looping behavior. So `w.syncUsingReconciliationStrategy` would have two here.